nixos/tests/nixos-test-driver/multi-os/qemu-vm.nix

```diff
@@ -0,0 +1,352 @@
+            description = "The mount point of the directory inside the virtual machine";
+          };
+          options.securityModel = mkOption {
+            type = types.enum [
+              "passthrough"
+              "mapped-xattr"
+              "mapped-file"
+              "none"
+            ];
+            default = "mapped-xattr";
+            description = ''
+              The security model to use for this share:
+
+              - `passthrough`: files are stored using the same credentials as they are created on the guest (this requires QEMU to run as root)
+              - `mapped-xattr`: some of the file attributes like uid, gid, mode bits and link target are stored as file attributes
+              - `mapped-file`: the attributes are stored in the hidden .virtfs_metadata directory. Directories exported by this security model cannot interact with other unix tools
+              - `none`: same as "passthrough" except the sever won't report failures if it fails to set file attributes like ownership
+            '';
+          };
+        }
+      );
+      default = { };
+      example = {
+        my-share = {
+          source = "/path/to/be/shared";
+          target = "/mnt/shared";
+        };
+      };
+      description = ''
+        An attributes set of directories that will be shared with the
+        virtual machine using VirtFS (9P filesystem over VirtIO).
+        The attribute name will be used as the 9P mount tag.
+      '';
+    };
+    # FIXME (copied from qemu-vm.nix)
+    virtualisation.interfaces = mkOption {
+      default = { };
+      example = {
+        enp1s0.vlan = 1;
+      };
+      description = ''
+        Network interfaces to add to the VM.
+      '';
+      type =
+        with types;
+        attrsOf (submodule {
+          options = {
+            vlan = mkOption {
+              type = types.ints.unsigned;
+              description = ''
+                VLAN to which the network interface is connected.
+              '';
+            };
+
+            assignIP = mkOption {
+              type = types.bool;
+              default = false;
+              description = ''
+                Automatically assign an IP address to the network interface using the same scheme as
+                virtualisation.vlans.
+              '';
+            };
+          };
+        });
+    };
+    system.name = mkOption {
+      type = types.str;
+      # FIXME: not accurate?
+      default = name;
+    };
+  };
+  config = {
+    # FIXME
+    system.build.vm =
+      hostPkgs.runCommand "nixos-vm"
+        {
+          preferLocalBuild = true;
+          meta.mainProgram = "run-${config.system.name}-vm";
+        }
+        ''
+          mkdir -p $out/bin
+          ln -s ${config.system.build.toplevel} $out/system
+          ln -s ${hostPkgs.writeScript "run-nixos-vm" startVM} $out/bin/run-${config.system.name}-vm
+        '';
+    virtualisation.qemu.options = [
+      "-enable-kvm" # ??
```

Comment:
Is this intended to run on Linux or on macOS?

Reply:
You're allowed to use Linux as a hypervisor of sorts on your Apple computer.

Reply:
I think these settings will not work on macOS hosts, though? That would significantly diminishes the usefulness given the fact that you’d need macOS to build a nix‐darwin configuration to run in the VMs to begin with. We have `darwin.linux-builder` to solve that bootstrapping problem one way, but it’s less tractable the other way around. It’s also not really practical to get cloud Macs running Linux.